keel-ec2-plugin/src/test/kotlin/com/netflix/spinnaker/keel/ec2/resource/SecurityGroupHandlerTests.kt

```diff
@@ -391,6 +392,44 @@ internal class SecurityGroupHandlerTests : JUnit5Minutests {
           .hasSize(cloudDriverResponse1.inboundRules.size + 1)
       }
     }
+
+    context("only inbound rule points to a non-existent security group") {
```

Comment:
As I mentioned above, I think if you create a response with more than 1 inbound rule, where one of them has a missing name for a referenced SG, then we'll ignore all other rules with your current implementation, which doesn't seem correct.

Reply:
@luispollo I updated the tests to try to show that the behavior doesn't skip subsequent rules. However, because the underlying collection is a set and not a list, the test doesn't guarantee the behavior is correct (I could just be getting lucky). But my other comment had a link to some code to demonstrate that the other rules don't get ignored.

Reply:
You could use `sortedSetOf`.

Reply:
TIL. Updated.

Reply:
Unfortunately, `sortedSetOf` didn't do what I thought it would. I thought it would preserve insertion ordering. Instead, it sorts the inputs (which should have been obvious based on the name). Since the security group rules aren't comparable, they can't be stored in a sorted set.

Reply:
Yeah, sorry, I didn't think to mentioned it, but there's a version of the function that takes a `Comparator` as a parameter, which is what I assumed you would use.

Reply:
Unfortunately, I'm not as smart as you think I am. 😝

Reply:
Yeah, right! 😆